### ISSUE TITLE
Synchronizing consumers

### ISSUE THREAD

**Issue by MaximeM** [millicast/millicast-player-unreal-engine-plugin#25]
Hello there,

First of all, super doc and plugin !

I want to diffuse in world on spatialized speakers the same track at the same timestamp, I thought that assigning each speaker as a consumer would do that.

![image](https://user-images.githubusercontent.com/1788688/188660599-8c7af2e8-772b-4a23-ba78-d39eca4257f3.png)

But no If I'm out of attenuation range the track just buffers and if I go out and go back after few seconds it resumes where it stopped.

I saw in source that there is a QueueAudioData in there maybe we could have the option to have less queue and an inSync stream between all consumers.

I thought it was fixed by @improbable-andreaskrugersen  in:
[Audio Sync PR](https://github.com/millicast/millicast-player-unreal-engine-plugin/pull/10)

Maybe I'm doing something wrong

**Comment by dbaldassi** [comment]
Hi,

So, if you are out of the audio range, the track just stops playing ? 
Even with only one consumer ?

I will investigates the options of the audio queue.

**Comment by dbaldassi** [comment]
Hi,
I have fixed the issue in f5eb9b1
We need to set the option ``PlayWhenSilent`` for the virtualization mode in the ``USoundWaveProcedural``

**Comment by MaximeM** [comment]
> Hi,
> 
> I have fixed the issue in f5eb9b1
> 
> We need to set the option ``PlayWhenSilent`` for the virtualization mode in the ``USoundWaveProcedural``

I'm going to try that thanks, I'll keep you updated.

**Comment by adrian-j-programmer** [comment]
Hi, I actually tested this extensively and we are still running into issues where audio can be severely out of sync with the video. This usually occurs if the audio consumer is very far away from the audio listener for a long while. In our case when you approach the stream that's on another floor, by the time you get to it the audio is at least several seconds behind the video. 
I'm experimenting now with disabling audio virtualization altogether, and will follow that experiment with resetting the sound wave procedural's audio queue when going out of the virtualized state.

**Comment by dbaldassi** [comment]
I noticed that the AudioComponent::IsVirtualized method returns true when we launch the game, and becomes false after ~2.5seconds which is the time of the audio lag.
And just looking at the BufferedAudioBytes }}in the {{SoundWaveProcedural , they do not decrease until the moment IsVirtualized becomes false, so the audio component does not actually play any audio until this moment.
So, I just added a condition to buffer audio data if the audio component is not virtualized.
```
void AMillicastAudioActor::QueueAudioData(const uint8* AudioData, int32 NumSamples)
{
if (!AudioComponent->IsVirtualized())

{ SoundStreaming->QueueAudio(AudioData, NumSamples * AudioParameters.GetNumberBytesPerSample()); }
}
```
